### PR TITLE
MCXN94X Adding LPTMR Counter Support

### PR DESCRIPTION
**Pull request by EmilioCBen**

### CODE REVIEW

File: dts/arm/nxp/nxp_mcxn94x_common.dtsi [hunk 1]
```diff
@@ -850,6 +850,15 @@
 		num-bidir-endpoints = <8>;
 		status = "disabled";
 	};
+
+	lptmr0: lptmr@4a000 {
+		compatible = "nxp,kinetis-lptmr";
```
Comment (butok) [inline dts/arm/nxp/nxp_mcxn94x_common.dtsi:855]: Can it be "nxp,lptmr" without kinetis?

File: boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.yaml [hunk 2]
```diff
@@ -27,4 +27,5 @@ supported:
   - regulator
   - adc
   - usb_device
+  - lpmtr
```
Comment (danieldegrasse) [inline boards/nxp/frdm_mcxn947/frdm_mcxn947_mcxn947_cpu0.yaml:30]: This yaml definition won't trigger any CI tests, as none have dependency on an IP name like `lptmr`. The tag you need to get counter tests to run is `counter`, which is already present

File: dts/arm/nxp/nxp_mcxn94x_common.dtsi (hunk 1 above)
Comment (danieldegrasse) [reply dts/arm/nxp/nxp_mcxn94x_common.dtsi:855]: Unfortunately no, not unless we change the compatible string across all parts. Typically we have left compatibles with the family name as is to avoid churn for downstream driver users, although we want to avoid adding new drivers with the family name (so this issue does not occur in the future)

Comment (butok) [reply dts/arm/nxp/nxp_mcxn94x_common.dtsi:855]: hope, we will be brave and clean it up sooner.

**Review by danieldegrasse — DISMISSED** [review]

**Review by butok — DISMISSED** [review]

File: dts/arm/nxp/nxp_mcxn94x_common.dtsi
Comment (danieldegrasse) [inline dts/arm/nxp/nxp_mcxn94x_common.dtsi:1]: Commit adding the LPTMR to MCX DTS should come after the commits renaming the LPTMR binding, and should use the new `nxp,lptmr` binding name

File: boards/nxp/frdm_mcxn947/board.c
Comment (danieldegrasse) [inline boards/nxp/frdm_mcxn947/board.c:1]: > boards: nxp: frdm_mcxn947

> Enabled one instance of lptmr
> running from the 16k FRO.

Commit title here needs to be more descriptive

File: drivers/timer/mcux_lptmr_timer.c
Comment (danieldegrasse) [inline drivers/timer/mcux_lptmr_timer.c:1]: This change should be in a separate commit from the updates to the counter driver

File: drivers/timer/Kconfig.mcux_lptmr [hunk 3]
```diff
@@ -6,9 +6,9 @@
 config MCUX_LPTMR_TIMER
 	bool "MCUX LPTMR timer"
 	default y
-	depends on DT_HAS_NXP_KINETIS_LPTMR_ENABLED
+	depends on DT_HAS_NXP_LPTMR_ENABLED
+	depends on !DT_HAS_NXP_OS_TIMER_ENABLED
```
Comment (decsny) [inline drivers/timer/Kconfig.mcux_lptmr:10]: this seems SOC specific maybe it should go in the SOC defconfig

**Review by danieldegrasse — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

File: drivers/counter/counter_mcux_lptmr.c [hunk 4]
```diff
@@ -210,7 +212,11 @@ static void mcux_lptmr_irq_config_0(const struct device *dev);
 
 static struct mcux_lptmr_config mcux_lptmr_config_0 = {
 	.info = {
+#if defined(CONFIG_SOC_SERIES_MCXNX4X)
+		.max_top_value = UINT32_MAX,
```
Comment (decsny) [inline drivers/counter/counter_mcux_lptmr.c:216]: Looks like this needs to come from DT

**Review by decsny — CHANGES_REQUESTED** [review]

File: dts/bindings/counter/nxp,lptmr.yaml [hunk 5]
```diff
@@ -29,3 +29,8 @@ properties:
   active-low:
     type: boolean
     description: Pulse counter input pin is active-low
+
+  max-top-value:
```
Comment (decsny) [inline dts/bindings/counter/nxp,lptmr.yaml:33]: might make more sense to put the resolution of the counter in bits

File: dts/bindings/counter/nxp,lptmr.yaml [hunk 6]
```diff
@@ -29,3 +29,8 @@ properties:
   active-low:
     type: boolean
     description: Pulse counter input pin is active-low
+
+  max-resolution:
```
Comment (decsny) [inline dts/bindings/counter/nxp,lptmr.yaml:33]: "max-resolution" does not make sense, just call it "resolution"

File: dts/bindings/counter/nxp,lptmr.yaml [hunk 7]
```diff
@@ -29,3 +29,8 @@ properties:
...
+    type: int
+    required: true
+    description: Represents the maximum width of the timer in bits.
```
Comment (decsny) [inline dts/bindings/counter/nxp,lptmr.yaml:36]: no maximum

**Review by decsny — CHANGES_REQUESTED** [review]

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by decsny** [comment]
reassigned to david because there is not really much of anything new here it's just an enablement of something already in tree on an nxp platform
